src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.c

```diff
@@ -806,18 +812,19 @@ static void glb_rr_connectivity_changed_locked(grpc_exec_ctx *exec_ctx,
                                                void *arg, grpc_error *error) {
   rr_connectivity_data *rr_connectivity = arg;
   glb_lb_policy *glb_policy = rr_connectivity->glb_policy;
-
   const bool shutting_down = glb_policy->shutting_down;
-  bool unref_needed = false;
   GRPC_ERROR_REF(error);
-
   if (rr_connectivity->state == GRPC_CHANNEL_SHUTDOWN || shutting_down) {
-    /* RR policy shutting down. Don't renew subscription and free the arg of
-     * this callback. In addition  we need to stash away the current policy to
-     * be UNREF'd after releasing the lock. Otherwise, if the UNREF is the last
-     * one, the policy would be destroyed, alongside the lock, which would
-     * result in a use-after-free */
-    unref_needed = true;
+    /* Either the RR policy or grpclb is shutting down: don't renew subscription
+     * and this callback's arg */
+    if (rr_connectivity->state == GRPC_CHANNEL_SHUTDOWN) {
+      /* An RR policy that has transitioned into the SHUTDOWN connectivity state
+       * should not be considered for picks or updates: the SHUTDOWN state is a
+       * sink, policies can't transition back from it. .*/
+      glb_policy->rr_policy_in_connectivity_shutdown = true;
```

Comment:
That was my first try. I've reintroduced it, with the extra tweaks needed to make it work.